content/en/docs/tasks/inject-data-application/distribute-credentials-secure.md

```diff
@@ -1,174 +1,174 @@
-* [Volume](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#volume-v1-core)
-* [Pod](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#pod-v1-core)
-
-{{% /capture %}}
-
-
+---
+title: Distribute Credentials Securely Using Secrets
+content_template: templates/task
+---
+
+{{% capture overview %}}
+This page shows how to securely inject sensitive data, such as passwords and
+encryption keys, using Deployments.
+{{% /capture %}}
+
+{{% capture prerequisites %}}
+
+{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}
+
+{{% /capture %}}
+
+{{% capture steps %}}
+
+## Convert your secret data to a base-64 representation
+
+Suppose you want to have two pieces of secret data: a username `my-app` and a password
+`39528$vdg7Jb`. First, use [Base64 encoding](https://www.base64encode.org/) to
+convert your username and password to a base-64 representation. Here's a Linux
+example:
+
+    echo -n 'my-app' | base64
+    echo -n '39528$vdg7Jb' | base64
+
+The output shows that the base-64 representation of your username is `bXktYXBw`,
+and the base-64 representation of your password is `Mzk1MjgkdmRnN0pi`.
+
+## Create a Secret
+
+Here is a configuration file you can use to create a Secret that holds your
+username and password:
+
+{{< code file="secret.yaml" >}}
+
+1. Create the Secret
+
+       kubectl create -f https://k8s.io/docs/tasks/inject-data-application/secret.yaml
+
+    {{< note >}}
+    **Note:** If you want to skip the Base64 encoding step, you can create a Secret
+    by using the `kubectl create secret` command:
+    {{< /note >}}
+
+       kubectl create secret generic test-secret --from-literal=username='my-app' --from-literal=password='39528$vdg7Jb'
+
+1. View information about the Secret:
+
+       kubectl get secret test-secret
+
+    Output:
+
+        NAME          TYPE      DATA      AGE
+        test-secret   Opaque    2         1m
+
+
+1. View more detailed information about the Secret:
+
+       kubectl describe secret test-secret
+
+    Output:
+
+        Name:       test-secret
+        Namespace:  default
+        Labels:     <none>
+        Annotations:    <none>
+
+        Type:   Opaque
+
+        Data
+        ====
+        password:   13 bytes
+        username:   7 bytes
+
+## Create a Deployment that has access to the secret data through a Volume
+
+Here is a configuration file you can use to create a Deployment:
+
+{{< code file="secret-deployment.yaml" >}}
+
+1. Create the Deployment:
+
+       kubectl create -f https://k8s.io/docs/tasks/inject-data-application/secret-deployment.yaml
+
+1. Verify that your Deployment is running:
+
+       kubectl get deployment secret-test
+
+    Output:
+
+        NAME               DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+        secret-test        2         2         2            2           12s
+
+1. Get a shell into the Container that is running in your Deployment:
+
+       kubectl exec -it secret-test -- /bin/bash

```

Comment:
There is no longer a Pod named secrete-test, so we'll have to change this command to something like `kubectl exec -it secret-test-6fdd57d45-q9wj5 -- /bin/bash`. Stand by, and I'll come up with a concrete suggestion.

Reply:
Here's what I think we should do:

`kubectl get pods`

The output shows the names of the Pods in your Deployment:

```
NAME                          READY     STATUS    RESTARTS   AGE
secret-test-6fdd57d45-q9wj5   1/1       Running   0          1m
secret-test-6fdd57d45-wtjtv   1/1       Running   0          1m
```

Get a shell into the Container that is running in one of your Pods:

`kubectl exec -it <pod-name> -- /bin/bash`

where `<pod-name>` is the name of one of your Pods.